src/common/procedure/src/error.rs

```diff
@@ -47,23 +48,20 @@ pub enum Error {
         backtrace: Backtrace,
     },
 
-    #[snafu(display("Failed to put {}, source: {}", key, source))]
-    PutState {
-        key: String,
-        source: object_store::Error,
-    },
+    #[snafu(display("Failed to put state, key: '{key}', error: {err_msg}"))]
+    PutState { key: String, err_msg: String },
```

Comment:
Just use BoxedError?
```suggestion
    PutState {
        key: String,
        #[snafu(backtrace)]
        source: BoxedError,
    },
```